ISISScientificComputing/autoreduce
# Refactor DataClean to handle 'Maintenance day(s)'

### Summary of work

- Added a method to convert a date string in a 'Maintenance day(s)' to a list of dates in the same format as Start/End
- Pulled out the code which formats a single date value from `date_formatter` into a separate `format_single_date` method. This enables the following:
- Added a condition for 'Maintenance day(s)' within `date_formatter` which formats >1 dates in the same format as Start/End, but as a list.

### How to test your work
- Run the `cycle_webscraper.py` script.
- Ensure that dates cleaned in the maintenance days column match the same date format as the start and end cycle dates columns.

### Additional comments
**Example**
_Input_:
![image](https://user-images.githubusercontent.com/31534888/75151639-165c0500-56ff-11ea-9c9b-592e86becdc8.png)
(etc.)

_Output_:
![image](https://user-images.githubusercontent.com/31534888/75151569-e7459380-56fe-11ea-9171-bf2170e23032.png)


Fixes #442

**Before merging ensure the release notes have been updated**
